### PR TITLE
post/2020-07-01-dev_and_iommu.md: post about IOMMU issue after SKINIT

### PR DESCRIPTION
**Pull request by krystian-hebel**
Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

### CODE REVIEW

File: blog/content/post/2020-07-01-dev_and_iommu.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+for the final test can be found [here](https://github.com/3mdeb/landing-zone/tree/test_iommu).
+
+As a proof that SLB protection works we took the fact that IOMMU is unable to
+read it, it is not possible to test it with the same ISA DMA engine because LZ
+is loaded into the higher than 1MB addresses. Unless otherwise specified, the
+DMA trial happens between each point below, as well as before and after the
+whole test. As a general rule, we tried to do the simplest possible protection,
+which is "deny all DMA".
+
+##### Test 1 (pass)
```
Comment (miczyg1) [inline blog/content/post/2020-07-01-dev_and_iommu.md:219]: @krystian-hebel I would add when we launch SKINIT in all these steps

File: blog/content/post/2020-07-01-dev_and_iommu.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+An important part of any trust chain is to make sure that the code that is run
+is the same code as was measured. This does not mean that the code must always
+be read-only, it can be a self-modifying code (and usually is, mostly because
+it performs some kind of relocation); it means that no external (as in: not
+measured) entity can change it, at least after the measurement is done.
+
+There are two main actors that can try to attack our TCB (Trusted Computing
+Base):
+
+* DMA-capable devices
```
[the remark this answers is not in the document]
Comment (krystian-hebel) [reply blog/content/post/2020-07-01-dev_and_iommu.md]: Done

File: blog/content/post/2020-07-01-dev_and_iommu.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+the cache is not invalidated and cannot be trusted.
+
+The test logic is simple: write some known values to memory, order legacy ISA
+DMA engine to overwrite it, read back that memory and compare with initial
+values. The memory range we choose is the first 20 bytes (21 actually, because
+of too safe `memset` function and implicit +1 added to DMA size) of memory,
+which is normally occupied by a real mode Interrupt Vector Table. This region
+was chosen because it must be in the first 1MB of memory (legacy DMA does not
+support higher addresses) and IVT is no longer used, so nothing bad happens if
+it gets corrupted. Initial value was written with `memset(_p(1), 0xcc, 0x20)`,
```
[the remark this answers is not in the document]
Comment (krystian-hebel) [reply blog/content/post/2020-07-01-dev_and_iommu.md:207]: My mind defaults to hex :) Fixed

File: blog/content/post/2020-07-01-dev_and_iommu.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+the value of the idle bus. The memory after the test is dumped twice to make
+sure that DMA engine had enough time to complete the transaction, even though
+printing through UART probably delays execution long enough already.
+
+The following tests were run on apu2, Fam17h does not support (or disables?)
+legacy ISA DMA. They use a variation of what is described under memory to memory
+DMA [here](http://www.osdever.net/tutorials/view/how-to-program-the-dma). Code
+for the final test can be found [here](https://github.com/3mdeb/landing-zone/tree/test_iommu).
+
+As a proof that SLB protection works we took the fact that IOMMU is unable to
```
[the remark this answers is not in the document]
Comment (krystian-hebel) [reply blog/content/post/2020-07-01-dev_and_iommu.md]: Done

File: blog/content/post/2020-07-01-dev_and_iommu.md (hunk 1 above)
Comment (krystian-hebel) [reply blog/content/post/2020-07-01-dev_and_iommu.md:226]: Added info in paragraph above, it is common for all tests

**Comment by miczyg1** [comment]
@artur-rs @KBanecka if you think it looks good you could publish it

**Pull request merged**